e2e-test-cloudbuild.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Google Cloud Builder CI configuration
+#
+
+steps:
+
+#
+# build the e2e test runner
+#
+
+- name: gcr.io/cloud-builders/docker
+  args: ['build', '-f', 'Dockerfile', '-t', 'e2e-runner', '.']
+  dir: 'build/e2e-image'
+  id: build-e2e
+  waitFor: ['-']
```

Comment:
Is there no way to share this image with a sub-build? Seems fine, just seems a bit redundant. :)

Reply:
Actually, there's a good reason to avoid this here: If you can move this step back to the main build, you can use `submit --no-source` and skip the complexity of the .gcloudignore file.

Can the name (`e2e-runner`) can be tagged and uploaded to the CI registry like the "normal" images?

Reply:
I can only think of one way to share is to push the image somewhere and pull it in the sub-build, since different cloud builds are running on different machines. But I feel it's a little overhead since this step takes only around 1 min.

Reply:
It's an interesting point that it might be possible to use `submit --no-source`. Does the `e2e-runner` image contains asset under `/test/e2e`? I think those source are necessary for the e2e test?

Reply:
Oh, you're right. I rescind that comment - you'd probably have to build the e2e binary and stuff it in the runner, which is more restructuring.